opentracing/ext/scope_manager/asyncio.py

```diff
@@ -0,0 +1,137 @@
+    that stores the :class:`~opentracing.Scope` in the current
+    :class:`Task` (:meth:`Task.current_task()`), falling back to
+    thread-local storage if none was being executed.
+
+    Automatic :class:`~opentracing.Span` propagation from
+    parent coroutines to their children is not provided, which needs to be
+    done manually:
+
+    .. code-block:: python
+
+        async def child_coroutine(span):
+            # activate the parent Span, but do not finish it upon
+            # deactivation. That will be done by the parent coroutine.
+            with tracer.scope_manager.activate(span, finish_on_close=False):
+                with tracer.start_active_span('child') as scope:
+                    ...
+
+        async def parent_coroutine():
+            with tracer.start_active_span('parent') as scope:
+                ...
+                await child_coroutine(span)
+                ...
+
+    """
+
+    def activate(self, span, finish_on_close):
+        """
+        Make a :class:`~opentracing.Span` instance active.
+
+        :param span: the :class:`~opentracing.Span` that should become active.
+        :param finish_on_close: whether *span* should automatically be
+            finished when :meth:`Scope.close()` is called.
+
+        If no :class:`Task` is being executed, thread-local
+        storage will be used to store the :class:`~opentracing.Scope`.
+
+        :return: a :class:`~opentracing.Scope` instance to control the end
+            of the active period for the :class:`~opentracing.Span`.
+            It is a programming error to neglect to call :meth:`Scope.close()`
+            on the returned instance.
+        """
+
+        task = self._get_task()
+        if not task:
+            return super(AsyncioScopeManager, self).activate(span,
+                                                             finish_on_close)
+
+        scope = _AsyncioScope(self, span, finish_on_close)
+        self._set_task_scope(scope, task)
+
+        return scope
+
+    @property
+    def active(self):
+        """
+        Return the currently active :class:`~opentracing.Scope` which
+        can be used to access the currently active
+        :attr:`Scope.span`.
+
+        :return: the :class:`~opentracing.Scope` that is active,
+            or ``None`` if not available.
+        """
+
+        task = self._get_task()
+        if not task:
+            return super(AsyncioScopeManager, self).active
+
+        return self._get_task_scope(task)
+
+    def _get_task(self):
+        try:
+            # Prevent failure when run from a thread
+            # without an event loop.
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
+            return None
+
+        return asyncio.Task.current_task(loop=loop)
+
+    def _set_task_scope(self, scope, task=None):
+        if task is None:
+            task = self._get_task()
+
+        setattr(task, '__active', scope)
```

Comment:
I'd say the mangling is enough here - but lets be safe and indeed use a `ot_` prefix here too. Will update.

Reply:
Yeah it's just to avoid any clash from whatever library is doing the same thing as our. Still internal with a clear "do not touch me!" flag :)

Good to go!